src/tests/testCacheManager.cc

```diff
@@ -17,11 +18,19 @@
 
 CPPUNIT_TEST_SUITE_REGISTRATION( testCacheManager );
 
+/// Provides test code access to CacheManager internal symbols
+class CacheManagerInternals : public CacheManager
+{
+public:
+    void ParseUrl(const AnyP::Uri &u) { CacheManager::ParseUrl(u); }
+};
+
```

Comment:
IMHO, it would be better to provide a CacheManager::TestParseUrl() public method than to add a wrapper class like this that will need to be maintained as CacheManager construction improves. Ideally, the unit test framework should provide a way to access private data members, but I am guessing that CPPUNIT is not that good.

I do not insist on any changes here.

Reply:
IMO this wrapper class is better. It keeps the nasty hack business restricted to the test case which requires it and avoids polluting the real class with accessors that might be abused.